test/e2e_node/runner/run_e2e.go

```diff
@@ -123,34 +148,57 @@ func WaitForUser() {
 	u.Done()
 }
 
-func runTests(fullhost string) ([]byte, error) {
+type TestResult struct {
+	fullhost               string
+	err                    error
+	testCombinedOutput     string
+	etcdCominedOutput      string
+	apiServerCominedOutput string
+	kubeletCominedOutput   string
+	varLogMessages		   string
+}
+
+func runTests(fullhost string) *TestResult {
+	result := &TestResult{fullhost: fullhost}
+
 	host := strings.Split(fullhost, ".")[0]
 	c := gcloud.NewGCloudClient(host, *zone)
 	// TODO(pwittrock): Come up with something better for bootstrapping the environment.
 	eh, err := c.RunAndWaitTillHealthy(
 		false, false, "4001", healthyTimeoutDuration, "v2/keys/", "etcd", "--data-dir", "./", "--name", "e2e-node")
-	defer func() { eh.TearDown() }()
+	defer func() {
+		eh.TearDown()
+		result.etcdCominedOutput = fmt.Sprintf("%s", eh.CombinedOutput.Bytes())
+	}()
 	if err != nil {
-		return nil, fmt.Errorf("Host %s failed to run command %v", host, err)
+		result.err = fmt.Errorf("Host %s failed to run command %v", host, err)
+		return result
 	}
 
 	apiBin := filepath.Join(kubeRoot, *kubeOutputRelPath, "kube-apiserver")
 	ah, err := c.RunAndWaitTillHealthy(
 		true, true, "8080", healthyTimeoutDuration, "healthz", apiBin, "--service-cluster-ip-range",
 		"10.0.0.1/24", "--insecure-bind-address", "0.0.0.0", "--etcd-servers", "http://127.0.0.1:4001",
-		"--v", "2", "--kubelet-port", "10250")
-	defer func() { ah.TearDown() }()
+		"--v", "2", "--alsologtostderr", "--kubelet-port", "10250")
+	defer func() {
+		ah.TearDown()
+		result.apiServerCominedOutput = fmt.Sprintf("%s", ah.CombinedOutput.Bytes())
+	}()
 	if err != nil {
-		return nil, fmt.Errorf("Host %s failed to run command %v", host, err)
+		result.err = fmt.Errorf("Host %s failed to run command %v", host, err)
+		return result
 	}
 
 	kubeletBin := filepath.Join(kubeRoot, *kubeOutputRelPath, "kubelet")
 	kh, err := c.RunAndWaitTillHealthy(
 		true, true, "10255", healthyTimeoutDuration, "healthz", kubeletBin, "--api-servers", "http://127.0.0.1:8080",
-		"--logtostderr", "--address", "0.0.0.0", "--port", "10250")
-	defer func() { kh.TearDown() }()
+		"--v", "2", "--alsologtostderr", "--address", "0.0.0.0", "--port", "10250")
```

Comment:
`--v=4` is recommended for debugging kubelet issues, but the log tends to be long and will be better uploaded somewhere.